pymc3/distributions/multivariate.py

```diff
@@ -1981,45 +2053,17 @@ class CAR(Continuous):
         "Generalized Hierarchical Multivariate CAR Models for Areal Data"
         Biometrics, Vol. 61, No. 4 (Dec., 2005), pp. 950-961
     """
+    rv_op = car
 
-    def __init__(self, mu, W, alpha, tau, sparse=False, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        D = W.sum(axis=0)
-        d, _ = W.shape
-
-        self.d = d
-        self.median = self.mode = self.mean = self.mu = at.as_tensor_variable(mu)
-        self.sparse = sparse
+    @classmethod
+    def dist(cls, mu, W, alpha, tau, sparse=False, *args, **kwargs):
 
         if not W.ndim == 2 or not np.allclose(W, W.T):
             raise ValueError("W must be a symmetric adjacency matrix.")
 
-        if sparse:
-            W_sparse = scipy.sparse.csr_matrix(W)
-            self.W = aesara.sparse.as_sparse_variable(W_sparse)
-        else:
-            self.W = at.as_tensor_variable(W)
-
-        # eigenvalues of D^−1/2 * W * D^−1/2
-        Dinv_sqrt = np.diag(1 / np.sqrt(D))
-        DWD = np.matmul(np.matmul(Dinv_sqrt, W), Dinv_sqrt)
-        self.lam = scipy.linalg.eigvalsh(DWD)
-        self.D = at.as_tensor_variable(D)
-
-        tau = at.as_tensor_variable(tau)
-        if tau.ndim > 0:
-            self.tau = tau[:, None]
-        else:
-            self.tau = tau
-
-        alpha = at.as_tensor_variable(alpha)
-        if alpha.ndim > 0:
-            self.alpha = alpha[:, None]
-        else:
-            self.alpha = alpha
+        return super().dist([mu, W, alpha, tau, sparse], **kwargs)
 
-    def logp(self, value):
+    def logp(value, mu, W, alpha, tau, sparse=False):
```

Comment:
There's no need for a `sparse` parameter here; plus, this option would only be useful to someone directly calling this function, which isn't too likely or common.  

More importantly, you can determine whether or not `W` is sparse by inspection.